specific_task/upload_iso/upload_iso_README.md

```diff
@@ -0,0 +1,158 @@
+# Hypercore ISO Uploader
+
+A Python script to download ISO images, verify their checksums, and upload them to Scale Computing Hypercore via the REST API.
+
+## Features
+
+- **Download**: Fetch ISOs from any HTTP/HTTPS URL with progress tracking
+- **Verify**: SHA256 checksum verification with support for multiple checksum file formats
+- **Upload**: Three-step upload process to Hypercore with detailed progress reporting
+- **Cleanup**: Automatic removal of temporary files after completion
+- **Flexible**: Supports environment variables or interactive prompts
+
+## Requirements
+
+- Python 3.6 or higher
+- `requests` library
+- Scale Computing Hypercore cluster with REST API access
+
+## Installation
+
+```bash
+pip install requests
+```
+
+For production use, consider using a virtual environment:
+
+```bash
+python3 -m venv venv
+source venv/bin/activate  # Linux/Mac
+pip install requests
+```
+
+## Usage
+
+### Interactive Mode
+
+Run the script and follow the prompts:
+
+```bash
+python3 upload_iso.py
+```
+
+You'll be prompted for:
+1. ISO download URL
+2. Checksum file URL
+3. Hypercore cluster IP/hostname
+4. Username and password
+
+### Environment Variables
+
+Set any combination of these to avoid repeated prompts:
+
+```bash
+# Credentials
+export HYPERCORE_CLUSTER="192.168.1.100"
```

Comment:
maybe nice to have / future enhancement - ansible and at least one other script up here uses these env vars - would be nice / convenient to standardize

SC_HOST=https://szt15b-01.lab.local
SC_PASSWORD=admin
SC_USERNAME=admin